dags/prio/processor.py

```diff
@@ -0,0 +1,330 @@
+bucket located in each server's project.
+
+The SubDag operator simplifies the definition of each phase. The clusters are
+ephemeral -- there are situations where clusters can become orphaned. These are
+usually resolved by rerunning the individual SubDag, which should clean-up
+properly.
+
+The following Airflow variables should be set:
+
+    prio_private_key_hex_internal
+    prio_public_key_hex_internal
+    prio_private_key_hex_external
+    prio_public_key_hex_external
+    prio_shared_secret
+
+These variables are encrypted and passed into the prio-processor container via
+the environment.
+"""
+from datetime import datetime, timedelta
+from os import environ
+
+from airflow import DAG
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.contrib.operators.gcs_to_gcs import (
+    GoogleCloudStorageToGoogleCloudStorageOperator,
+)
+from airflow.operators import PythonOperator
+from airflow.operators.subdag_operator import SubDagOperator
+from prio import dataproc, kubernetes
+
+DEFAULT_ARGS = {
+    "owner": "amiyaguchi@mozilla.com",
+    "depends_on_past": True,
+    "wait_for_downstream": True,
+    "start_date": datetime(2019, 8, 22),
+    "email": [
+        "amiyaguchi@mozilla.com",
+        "hwoo@mozilla.com",
+        "dataops+alerts@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 0,
+    "retry_delay": timedelta(minutes=15),
+    "schedule_interval": "@daily",
+    "dagrun_timeout": timedelta(hours=2),
+}
+
+# use a less than desirable method of generating the service account name
+IS_DEV = environ.get("DEPLOY_ENVIRONMENT") == "dev"
+ENVIRONMENT = "dev" if IS_DEV else "prod"
+
+PROJECT_ADMIN = GoogleCloudStorageHook("google_cloud_prio_admin").project_id
+PROJECT_A = GoogleCloudStorageHook("google_cloud_prio_a").project_id
+PROJECT_B = GoogleCloudStorageHook("google_cloud_prio_b").project_id
+
+SERVICE_ACCOUNT_ADMIN = "prio-admin-runner@{}.iam.gserviceaccount.com".format(
+    PROJECT_ADMIN
+)
+SERVICE_ACCOUNT_A = "prio-runner-{}-a@{}.iam.gserviceaccount.com".format(
+    ENVIRONMENT, PROJECT_A
+)
+SERVICE_ACCOUNT_B = "prio-runner-{}-b@{}.iam.gserviceaccount.com".format(
+    ENVIRONMENT, PROJECT_B
+)
+
+BUCKET_PRIVATE_A = "moz-fx-prio-{}-a-private".format(ENVIRONMENT)
+BUCKET_PRIVATE_B = "moz-fx-prio-{}-b-private".format(ENVIRONMENT)
+BUCKET_SHARED_A = "moz-fx-prio-{}-a-shared".format(ENVIRONMENT)
+BUCKET_SHARED_B = "moz-fx-prio-{}-b-shared".format(ENVIRONMENT)
+BUCKET_DATA_ADMIN = "moz-fx-data-{}-prio-data".format(ENVIRONMENT)
+BUCKET_BOOTSTRAP_ADMIN = "moz-fx-data-{}-prio-bootstrap".format(ENVIRONMENT)
+
+dag = DAG(dag_id="prio_processor", default_args=DEFAULT_ARGS)
+
+
+# Copy the dependencies necessary for running the `prio-processor staging` job
+# into the relevant GCS locations. This includes the runner and the egg
+# containing the bundled dependencies.
+
+prio_staging_bootstrap = SubDagOperator(
```

Comment:
The use of the word staging here is confusing. The top of file comment doesn't mention a staging job.

Reply:
Used staging instead of partitioning to keep the terminology consistent. If this gets changed, it should be changed in the processor job so things are consistent. I do agree that the naming could be better though.